File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0 # Edge case for empty list

        unique_index = 0 # Slow pointer tracking unique element position

        for i in range(1, len(nums)): # Fast pointer iterating through array
            if nums[i] != nums[unique_index]: # Found a new unique element
                unique_index += 1 # Move slow pointer forward
                nums[unique_index] = nums[i] # Overwrite with the new unique value
        
        return unique_index + 1 # The count of unique elements

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.c
int removeDuplicates(int* nums, int numsSize) {
    int write = 1;

    for (int i = 1; i < numsSize; i++) {
        if (nums[i] != nums[write - 1]) {
            nums[write] = nums[i];
            write++;
        }
    }

    return write;
}